ceno_zkvm/src/instructions.rs

```diff
@@ -47,31 +47,31 @@ pub trait Instruction<E: ExtensionField> {
         config: &Self::InstructionConfig,
         num_witin: usize,
         steps: Vec<StepRecord>,
-    ) -> Result<(RowMajorMatrix<E::BaseField>, LkMultiplicity), ZKVMError> {
+    ) -> Result<(RowMajorMatrix<E::BaseField>, Vec<LkMultiplicity>), ZKVMError> {
         let nthreads = max_usable_threads();
         let num_instance_per_batch = if steps.len() > 256 {
             steps.len().div_ceil(nthreads)
         } else {
             steps.len()
         }
         .max(1);
-        let lk_multiplicity = LkMultiplicity::default();
         let mut raw_witin = RowMajorMatrix::<E::BaseField>::new(steps.len(), num_witin);
         let raw_witin_iter = raw_witin.par_batch_iter_mut(num_instance_per_batch);
 
-        raw_witin_iter
+        let lk_multiplicity = raw_witin_iter
             .zip(steps.par_chunks(num_instance_per_batch))
             .flat_map(|(instances, steps)| {
-                let mut lk_multiplicity = lk_multiplicity.clone();
                 instances
                     .chunks_mut(num_witin)
                     .zip(steps)
                     .map(|(instance, step)| {
+                        let mut lk_multiplicity = LkMultiplicity::default();
```

Comment:
This new design although idea is nice and capture lkm on row (per instance) level, however this design will bring quite lot of overhead because we will create #instance of lkm, comparing with original design we just create ~ #opcode + #opcode lkm which is < 50.

So I suggest we just rollback to single lkm version. In current assignment design, lkm check failed will imply on all row contributed to the failed, because we dont have per row conditional logic. With single lkm, we can still know which table failed, while retain the efficiency of the design